common.sh

```diff
@@ -126,6 +123,19 @@ function fetch_tools {
     curl -sL ${kb_tools_download_url} -o "$kb_tools_archive_path"
   fi
   tar -zvxf "$kb_tools_archive_path" -C "$tmp_root/"
+
+  header_text "ensuring dep is present for legacy v1 scaffold tests"
+  if ! which dep; then
+    mkdir -p test_gobin
+    # can't install dep with modules due to dependency versioning (lol)
+    curl -sL -o ${test_gobin}/dep https://github.com/golang/dep/releases/download/0.5.2/dep-$(go env GOOS)-$(go env GOARCH)
+  fi
+
+  header_text "ensuring kind is present for e2e tests"
+  if ! which kind; then
+    mkdir -p test_gobin
+    GOBIN=${test_gobin} GO111MODULE="on" go get sigs.k8s.io/kind@v0.3.0
```

Comment:
This causes a diff on the projects root-level go.mod, which is not expected? Does go literally need us to `go get` outside of _any_ project? I'm too new to golang (and modules) to have an opinion, and I'm looking forward to knowing what the go tooling expects from users here...

```
diff --git a/go.mod b/go.mod
index bbb594bf..0163e4b6 100644
--- a/go.mod
+++ b/go.mod
@@ -4,7 +4,6 @@ go 1.12
 
 require (
        github.com/gobuffalo/envy v1.6.15 // indirect
-       github.com/inconshreveable/mousetrap v1.0.0 // indirect
        github.com/markbates/inflect v1.0.4
        github.com/onsi/ginkgo v1.8.0
        github.com/onsi/gomega v1.5.0
@@ -12,9 +11,9 @@ require (
        github.com/spf13/afero v1.2.2
        github.com/spf13/cobra v0.0.3
        github.com/spf13/pflag v1.0.3
-       golang.org/x/net v0.0.0-20190514140710-3ec191127204 // indirect
        golang.org/x/sys v0.0.0-20190506115046-ca7f33d4116e // indirect
        golang.org/x/text v0.3.2 // indirect
        golang.org/x/tools v0.0.0-20190506145303-2d16b83fe98c
        gopkg.in/yaml.v2 v2.2.2
+       sigs.k8s.io/kind v0.3.0 // indirect
 )
```

Reply:
go tooling likes modifying everything.  I think there may be a flag we can pass to tell it to not do this.

Reply:
hmm... looked around a bit, maybe not.  can run `go mod tidy` afterwards to fix it.